From 389e9eb9644452d5d88f5d48553a61b15e93fb75 Mon Sep 17 00:00:00 2001
From: Stuart MENEFY <stuart.menefy@st.com>
Date: Tue, 14 Oct 2008 19:46:15 +0100
Subject: [PATCH] sh_stm: Fix errors in the STx7105 interrupt configuration table

The STx7105 interrupt table was missing some entries because of
documentation problems. Also fix some spelling mistakes.

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
Signed-off-by: David McKay <david.mckay@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7105.c |   19 ++++++++++---------
 1 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7105.c b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
index 88b9eb6..43932da 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
@@ -1202,12 +1202,12 @@ enum {
 	TVO_DCS0, NAND, DELMU_PP, DELMU_MBE,		/* Group 10 */
 	MAIN_VDP_FIFO_EMPTY, MAIN_VDP_END_PROCESSING,	/* Group 11 */
 	MAIN_VTG, AUX_VTG,
-	HDMI_CEC_WAKEUP, HDMA_CEC, HDMI, HDCP,		/* Group 12 */
+	HDMI_CEC_WAKEUP, HDMI_CEC, HDMI, HDCP,		/* Group 12 */
 	PTI0, PDES_ESA, PDES, PDES_READ_CW,		/* Group 13 */
 	TKDMA_TKD, TKDMA_DMA, CRIPTO_SIGDMA,		/* Group 14 */
 	CRIPTO_SIG_CHK,
 	OHCI0, EHCI0, TVO_DCS1, BDISP_CQ,		/* Group 15 */
-	ICMA3_KTE, ICAM3, KEY_SCANNER, MES,		/* Group 16 */
+	ICAM3_KTE, ICAM3, KEY_SCANNER, MES,		/* Group 16 */
 
 	/* interrupt groups */
 	TMU2, RTC,
@@ -1257,7 +1257,7 @@ static struct intc_vect vectors[] = {
 	INTC_VECT(DELMU_PP, 0x14c0), INTC_VECT(DELMU_MBE, 0x14e0),
 	INTC_VECT(MAIN_VDP_FIFO_EMPTY, 0x1500), INTC_VECT(MAIN_VDP_END_PROCESSING, 0x1520),
 	INTC_VECT(MAIN_VTG, 0x1540), INTC_VECT(AUX_VTG, 0x1560),
-	INTC_VECT(HDMI_CEC_WAKEUP, 0x1580), INTC_VECT(HDMA_CEC, 0x15a0),
+	INTC_VECT(HDMI_CEC_WAKEUP, 0x1580), INTC_VECT(HDMI_CEC, 0x15a0),
 	INTC_VECT(HDMI, 0x15c0), INTC_VECT(HDCP, 0x15e0),
 	INTC_VECT(PTI0, 0x1600), INTC_VECT(PDES_ESA, 0x1620),
 	INTC_VECT(PDES, 0x1640), INTC_VECT(PDES_READ_CW, 0x1660),
@@ -1265,7 +1265,7 @@ static struct intc_vect vectors[] = {
 	INTC_VECT(CRIPTO_SIGDMA, 0x16c0), INTC_VECT(CRIPTO_SIG_CHK, 0x16e0),
 	INTC_VECT(OHCI0, 0x1700), INTC_VECT(EHCI0, 0x1720),
 	INTC_VECT(TVO_DCS1, 0x1740), INTC_VECT(BDISP_CQ, 0x1760),
-	INTC_VECT(ICMA3_KTE, 0x1780), INTC_VECT(ICAM3, 0x17a0),
+	INTC_VECT(ICAM3_KTE, 0x1780), INTC_VECT(ICAM3, 0x17a0),
 	INTC_VECT(KEY_SCANNER, 0x17c0), INTC_VECT(MES, 0x17e0),
 };
 
@@ -1290,11 +1290,12 @@ static struct intc_group groups[] = {
 	INTC_GROUP(GROUP10, TVO_DCS0, NAND, DELMU_PP, DELMU_MBE),
 	INTC_GROUP(GROUP11, MAIN_VDP_FIFO_EMPTY, MAIN_VDP_END_PROCESSING,
 		   MAIN_VTG, AUX_VTG),
-	INTC_GROUP(GROUP12, HDMI_CEC_WAKEUP, HDMA_CEC, HDMI, HDCP),
+	INTC_GROUP(GROUP12, HDMI_CEC_WAKEUP, HDMI_CEC, HDMI, HDCP),
 	INTC_GROUP(GROUP13, PTI0, PDES_ESA, PDES, PDES_READ_CW),
-	INTC_GROUP(GROUP15, TKDMA_TKD, TKDMA_DMA, CRIPTO_SIGDMA,
+	INTC_GROUP(GROUP14, TKDMA_TKD, TKDMA_DMA, CRIPTO_SIGDMA,
 		   CRIPTO_SIG_CHK),
-	INTC_GROUP(GROUP16, OHCI0, EHCI0, TVO_DCS1, BDISP_CQ),
+	INTC_GROUP(GROUP15, OHCI0, EHCI0, TVO_DCS1, BDISP_CQ),
+	INTC_GROUP(GROUP16, ICAM3_KTE, ICAM3, KEY_SCANNER, MES),
 };
 
 static struct intc_prio priorities[] = {
@@ -1334,11 +1335,11 @@ static struct intc_mask_reg mask_registers[] = {
 	    SSC0, SSC1, SSC2, SSC3, 				/*  7...4 */
 	    PIO3, PIO4, PIO5,  PIO6  } },			/*  3...0 */
 	{ 0x00000348, 0x00000368, 32, /* INTMSK08 / INTMSKCLR08 */
-	  { MES, KEY_SCANNER, ICAM3, ICMA3_KTE,			/* 31..28 */
+	  { MES, KEY_SCANNER, ICAM3, ICAM3_KTE,			/* 31..28 */
 	    BDISP_CQ, TVO_DCS1, EHCI0, OHCI0,			/* 27..24 */
 	    CRIPTO_SIG_CHK, CRIPTO_SIGDMA, TKDMA_DMA, TKDMA_TKD,/* 23..20 */
 	    PDES_READ_CW, PDES, PDES_ESA, PTI0,			/* 19..16 */
-	    HDCP, HDMI, HDMA_CEC, HDMI_CEC_WAKEUP,		/* 15..12 */
+	    HDCP, HDMI, HDMI_CEC, HDMI_CEC_WAKEUP,		/* 15..12 */
 	    AUX_VTG, MAIN_VTG, MAIN_VDP_END_PROCESSING,		/* 11...8 */
 		 MAIN_VDP_FIFO_EMPTY,
 	    DELMU_MBE, DELMU_PP, NAND, TVO_DCS0,		/*  7...4 */
-- 
1.5.3.6

